File: Python/count_subarr_sum_k.py
n = int(input())
nums = []
for _ in input().split():
    nums.append(int(_))

k = int(input())
presum = {}
sum = 0
count = 0
for i in range(n):
    sum += nums[i]
    if sum == k:
        count += 1
    rem = sum - k
    if rem in presum:
        count += presum[rem]
    if sum not in presum:   
        presum.setdefault(sum, 1)
    else:
        presum[sum] += 1

print(count)

File: C++/count_subarr_sum_k.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    int n;
    cin >> n;
    vector<int> nums(n);
    for(int i = 0; i < n; i++) {
        cin >> nums[i];
    }
    int k;
    cin >> k;
    unordered_map<int, int> presum;
    int count = 0;
    int sum = 0;
    for(int i = 0; i < n; i++) {
        sum += nums[i];
        if(sum == k) {
            count++;
        }
        if(presum.find(sum - k) != presum.end()) {
            count += presum[sum - k];
        }
        presum[sum]++;
    }
    cout << count << endl;
    return 0;
}